galsim/base.py

```diff
@@ -1462,10 +1462,12 @@ def drawKImage(self, re=None, im=None, nx=None, ny=None, bounds=None, scale=None
 
         return re,im
 
-    # Quick and dirty.  Just check reprs are equal.
-    def __eq__(self, other): return repr(self) == repr(other)
+    def __eq__(self, other):
```

Comment:
Why do we check only these things and not, say, the noise attribute (if it has one)?  If two objects have the same SBProfile and type GSObject, but different ```noise``` attribute, should they be considered equal?